manifests/system_table.pp

```diff
@@ -0,0 +1,65 @@
+# Add a system table $name to /etc/incron.d
+#
+# @param name
+#   The name of the table in /etc/incron.d/
+# @param path
+#   Filesystem path to monitor
+# @param mask
+#   Symbolic array or numeric mask for events
+# @param command
+#   Command to run on detection of event in $path
+# @param custom_content
+#   Custom content to add to /etc/incron.d/$name
+define incron::system_table (
+  Optional[Stdlib::AbsolutePath] $path    = undef,
+  Array[String] $mask                     = ['IN_MODIFY','IN_MOVE','IN_CREATE','IN_DELETE'],
+  Optional[Stdlib::AbsolutePath] $command = undef,
+  Optional[String] $custom_content        = undef
+) {
+  include '::incron'
+
+  if !$path and !$command and !$custom_content {
+    fail ('You must specify either $path and $command or $custom_content.')
```

Comment:
Based on the man page, I believe the message is correct and the logic is not.  The logic should be  ($path and $command) or $custom_content.  This will impact the test.  Also, I don't really understand the use case for the custom_content, as the man page specifies each line to be 'path mask command'.  In other words, we give the user everything they need to build a valid line.